### DISCUSSION TITLE
Passing template variable to GrpcOperator

### DISCUSSION THREAD

**Discussion by abecquet77** [apache/airflow#36661] [Q&A]
I can't pass template variables into GrpcOperator data.

Example :
```
                GrpcOperator(
                        task_id=f"call_grpc",
                        grpc_conn_id="grpc_01",
                        custom_connection_func=grpc_connection,
                        call_func="RunQuery",
                        stub_class=ApiStub,
                        data={'request': RunQueryRequest(start_date="{{ data_interval_end) }}",stop_date="{{ data_interval_end) }}")},
                    )
```

It is possible to do that ?

**Comment by josh-fell** [comment]
Probably not in that form.
1. The call to `RunQueryRequest`, whatever it is, will be run during parse time, and I assume is using the literal strings of "{{ data_interval_end }}"? 
2. You do have a typo in your code snippet with the extra parentheses after "_end".

[Defining a `user_defined_macro`](https://docs.astronomer.io/learn/templating#macros-using-custom-functions-and-variables-in-templates) might help you here in this case if the typo fix doesn't resolve the issue.

**Comment by abecquet77** [comment]
When we call grpc with Python we have to use a function Request to passing message.
So it's impossible to use Jinja Template in data argument for GrpcOperator

**Comment by mcpeanutbutter** [comment]
I got stuck at the same issue and feared that this would render the `GrpcOperator` pretty useless for my project. However, there's a workaround: You can implement a custom operator based on the `GrpcOperator` that populates the `data` field at `execute` time. I have documented this in a stack overflow question: https://stackoverflow.com/a/78477317/11831200
